3_descriptives/Descriptives_lesson_key.Rmd

```diff
@@ -7,223 +7,141 @@ output:
-# you can exclude them if you want with a simple argument 
-# notice that the y axis also automatically changes!
-boxplot(df$mmi, outline = F)
+# how many subjects are in this dataframe?
+Nsubs <- length(SubList)
 
-# what if you prefer horizonal plots? 
-boxplot(df$mmi, horizontal = TRUE)
 ```
 
-#3B. Boxplots with grouping 
-We can also use grouping variables in boxplots. Here we will plot ADHD symptoms by group: high or low media multitasking (HMM or LMM)
+## Make sure you know how many trials there were per subj.
 ```{r}
-#mmi = y variable, groupStatus is our (x) variable
-# we are only interested in the high multitasking and and low multitasking groups, so we will filter the data to remove those in the middle (and therefore not labeled for groupStatus)
-
-df.group <- subset(df, groupStatus != "")
-df.group$groupStatus <- as.factor(df.group$groupStatus)
-boxplot(df$bis ~  df$groupStatus, data=df)
-#*** PAUL HOW TO GET RID OF 3rd empty LEVEL?
-  
-# let's make this one prettier by adding axis labels and colors 
-boxplot(adhd ~ groupStatus, data=df.group,
-  	xlab="Multitasking Group", ylab="ADHD", col = c("light blue", "light green"))
+df %>%
+  group_by (subjNum) %>%
+  summarize (nTrials = n())
+
 ```
 
-# 5. Relationships between two continuous variables
+# 6. saving your work
 
+## Saving tables into csv files 
 ```{r}
-# this is the most basic plot you can make. 
-#don't worry, later we will show you how to make much prettier graphs! 
+ basic_descriptives <-  df %>% 
+   # if adhd score is lower than mean, label "low", else label "high""
+   mutate(adhdF = ifelse(adhd < mean(adhd), "Low", "High")) %>%
+  # split the summary statistics by group 
+  group_by(adhdF, groupStatus) %>%
+  summarize(meanHitRate = mean(hitRate, na.rm= T), 
+            meanFalseAlarm= mean(faRate, na.rm= T),
+            meanDprime = mean(dprime, na.rm = T))
 
-# graph the relationship between ADHD (x) and impulsivity (y)
-plot (df$adhd, df$bis)
-```
+# use write.csv() to save this out!
+write.csv(basic_descriptives, file = "myDescriptives.csv")
 
-# 6. Relationships of many variables
-We can use scatterplot matrices  which are a handy & quick way of visualizing relationships between multiple variables in a dataset. This works best when you have 3-5 variables (it gets too hard to read after that).
-Let's look at the relationships between multitasking, impulsitivity, and adhd symptoms.
 
-```{r}
-ncol(df)
-
-# We will use the pairs function 
-# in order to isolate the 3 variables we want, indexing [16:18] variables 
-pairs (df[16:18])
-# There are positive relationships evident between each pair of variables! 
-# We will explore these relationshisp more in the statistics tutorial.
+# note : it will automatically save to your working directory unless you specify otherise 
+# remember to check your working directory, use gwd() 
+# to change your working directory, use setwd() 
+# or, add the path manually "/dir/dir/dir/filename.csv"
```

Comment:
I would only include the suggestion to add the path manually. Let's try to avoid teaching people to change the working directory, to help with between-computer reproducibility of code!